tests/integration/test_array.py

```diff
@@ -24,10 +27,147 @@ def main(xs: array[float, 42]) -> int:
     assert val.val.v == 42
 
 
-@pytest.mark.skip("Skipped until Hugr lowering is updated")
 def test_index(validate):
     @compile_guppy
     def main(xs: array[int, 5], i: int) -> int:
         return xs[0] + xs[i] + xs[xs[2 * i]]
 
     validate(main)
+
+
+def test_subscript_drop_rest(validate):
+    module = GuppyModule("test")
+    module.load(quantum)
+
+    @guppy.declare(module)
+    def foo() -> array[int, 10]: ...
+
+    @guppy(module)
+    def main() -> int:
+        return foo()[0]
+
+    validate(module.compile())
+
+
+def test_linear_subscript(validate):
+    module = GuppyModule("test")
+    module.load(quantum)
+
+    @guppy.declare(module)
+    def foo(q: qubit @inout) -> None: ...
+
+    @guppy(module)
+    def main(qs: array[qubit, 42], i: int) -> array[qubit, 42]:
+        foo(qs[i])
+        return qs
+
+    validate(module.compile())
+
+
+def test_inout_subscript(validate):
+    module = GuppyModule("test")
+    module.load(quantum)
+
+    @guppy.declare(module)
+    def foo(q: qubit @inout) -> None: ...
+
+    @guppy(module)
+    def main(qs: array[qubit, 42] @inout, i: int) -> None:
+        foo(qs[i])
```

Comment:
both this and the example above would fail if the argument to `foo` wasn't inout right?

Reply:
Yes 👍